### DISCUSSION TITLE
How to set custom path for files outside the project directory

### DISCUSSION THREAD

**Discussion by haripriya-goswami** [adonisjs/core#2807] [Help] [answered]
I want to download a file from the adonis(v5) application in a directory outside the project directory. How can I make it possible? Please help.

**Comment by thetutlage** [comment, marked as answer]
Please share the code you have written so far when trying to download the file outside of the project directory

**Reply by haripriya-goswami** [reply]
So, as far I've been able to run `response.attachment(...)` to download a file in the default downloads directory on the system but I want to download it in a specified directory path outside the project directory. I thought of using the file system ([fs-extra](https://www.npmjs.com/package/fs-extra)) explicitly but getting errors in importing that package in node script as well.

**Reply by thetutlage** [reply]
Hard to help unless you share complete information. What error have you received, what arguments have to passed to `response.attachment`. Imaginary situations cannot be answered

**Reply by haripriya-goswami** [reply]
This is the code nugget that's downloading the file in the default downloads directory on the system.
```
let filePath = Application.publicPath(`assets/${profile}.webp`
response.attachment(filePath, `${profile}`, 'attachment')
```

**Reply by thetutlage** [reply]
So, all you need to do is not use the `Application.publicPath` and get the absolute path to where the file is stored. It can be anywhere on the disk.

```ts
let filePath = '/absolute/path/to/the/file/on/the/server'
response.attachment(filePath, `${profile}`, 'attachment')
```
